src/hotspot/share/gc/z/zDirector.cpp

```diff
@@ -535,7 +535,8 @@ static double calculate_young_to_old_worker_ratio(const ZDirectorStats& stats) {
   const size_t reclaimed_per_old_gc = stats._old_stats._stat_heap._reclaimed_avg;
   const double current_young_bytes_freed_per_gc_time = double(reclaimed_per_young_gc) / double(young_gc_time);
   const double current_old_bytes_freed_per_gc_time = double(reclaimed_per_old_gc) / double(old_gc_time);
-  const double old_vs_young_efficiency_ratio = current_old_bytes_freed_per_gc_time / current_young_bytes_freed_per_gc_time;
+  const double old_vs_young_efficiency_ratio = current_young_bytes_freed_per_gc_time == 0 ? std::numeric_limits<double>::infinity()
+                                                                                          : current_old_bytes_freed_per_gc_time / current_young_bytes_freed_per_gc_time;
```

Comment:
I returning infinity here will cause problems with NaN down the line. It is also unclear what this means if both are `0`. To me something like the following makes sense. But I will discus this with my team.
```suggestion

  if (current_young_bytes_freed_per_gc_time == 0.0) {
    if (current_old_bytes_freed_per_gc_time == 0.0) {
      // Neither young nor old collections have reclaimed any memory.
      // Give them equal priority.
      return 1.0;
    }

    // Only old collections have reclaimed memory.
    // Prioritize old.
    return ZOldGCThreads;
  }

```

Reply:
Hi Axel, thanks for the suggestion.
Please discuss it in your team and tell me the outcome :-) !

Reply:
I think you can go ahead and use my proposed change. And I will review it. 

@fisk, who has the most familiarity with the heuristics code, is currently on vacation. So we should wait until he is back and let him review this as well.